src/main/java/org/elasticsearch/index/mapper/core/DateFieldMapper.java

```diff
@@ -253,6 +258,15 @@ public void setTimeUnit(TimeUnit timeUnit) {
             this.dateMathParser = new DateMathParser(dateTimeFormatter);
         }
 
+        public boolean isIndexCreatedBefore2x() {
```

Comment:
I've tried keeping this kind of 2x stuff out of the field types.  Instead, could we have a 2x formatter that wraps the other formatter, and does this fallback? Then we can only add the wrapper when version < 2x (which we have when building).